ext/test/upgrade-and-exit

```diff
@@ -61,17 +61,21 @@ test ! -e "$PDBBOX"/var/mq-migrated
 psql -U puppetdb puppetdb -c '\dt' >"$tmpdir/out" 2>"$tmpdir/err"
 cat "$tmpdir/out"
 cat "$tmpdir/err"
+
 # Output moved to err and changed as of at least pg 11
 grep 'No relations found' "$tmpdir/out" \
     || grep 'Did not find any relations' "$tmpdir/err"
 
+psql -U $PDB_TEST_DB_ADMIN puppetdb -c 'create extension if not exists pg_trgm'
+psql -U $PDB_TEST_DB_ADMIN puppetdb -c 'create extension if not exists pgcrypto'
+
```

Comment:
PE has a requirement of pgcrypto - see use of `md5_agg` there. Why wouldn't we want to make pgcrypto required? The use of it here makes the migration dramatically faster than it would be otherwise. Considering the size of the events table, I think this is worthwhile.

Reply:
Even if we decide we want to, I suspect we actually can't until perhaps an X release -- i.e. I'm currently inclined toward thinking we wouldn't want to require changes like that to the production database in a Z (or even Y?).